foodbornenyc/sources/twitter_search.py

```diff
@@ -154,8 +161,10 @@ def query_twitter(how_long=0, interval=5):
             continue
         #logger.info("%i Total unique tweets in %i:%i:%i time", len(id_set),
         #            *sec_to_hms(time()-start))
-        new_tweets = [ t for t in tweets if t['id_str'] not in id_set ]
-        new_Tweets = tweets_to_Tweets(new_tweets, fields)
+        new_tweets = [t for t in tweets if t['id_str'] not in id_set]
+        # if a retrieved tweet has a loc/user with a matching ID already in the
+        # db, that loc/user is updated instead of a new one added, bc of merge
+        new_Tweets = [db.merge(t) for t in tweets_to_Tweets(new_tweets, fields)]
```

Comment:
Hmm ok.  By that do you mean both tweets have a reference to the same `user`? Because I think that's automatically resolvable.  To test this is simple, try it without using the id set as a filter and try merging two duplicate tweets in a transaction.  If this works fine, then using just merge is ok. 

Only reason I'm skeptical is because I think I remember having to add the id set because I was getting errors due to duplicate tweet ids. But I don't really remember so let's just find out.

Also doing some manual bookkeeping may be faster -- or not -- maybe worth timing.

Oh but wait now I checked that link and it says merge consults the local memory map.  Let's just test it and know for sure :)